integrations/gitlab_v1/main.py

```diff
@@ -0,0 +1,196 @@
+    """Handle resynchronization for GitLab projects."""
+    if kind != ObjectKind.PROJECT:
+        logger.warning(f"Unexpected kind {kind} for on_resync_projects")
+        return []
+
+    if not hasattr(ocean, 'gitlab_handler'):
+        logger.error("GitLab handler not initialized. Please check on_start function.")
+        return []
+
+
+    fetch_method = lambda: ocean.gitlab_handler.fetch_resources(ObjectKind.PROJECT)
+    items = await fetch_resource(fetch_method)
+    return items
+
+
+@ocean.on_resync(ObjectKind.MERGE_REQUEST)
+async def on_resync_merge_requests(kind: str) -> ASYNC_GENERATOR_RESYNC_TYPE:
+    """Handle resynchronization for GitLab merge requests."""
+    if kind != ObjectKind.MERGE_REQUEST:
+        logger.warning(f"Unexpected kind {kind} for on_resync_merge_requests")
+        return []
+
+    if not hasattr(ocean, 'gitlab_handler'):
+        logger.error("GitLab handler not initialized. Please check on_start function.")
+        return []
+
+
+    fetch_method = lambda: ocean.gitlab_handler.fetch_resources(ObjectKind.MERGE_REQUEST)
+    items = await fetch_resource(fetch_method)
+    return items
+
+
+@ocean.on_resync(ObjectKind.ISSUE)
+async def on_resync_issues(kind: str) -> ASYNC_GENERATOR_RESYNC_TYPE:
+    """Handle resynchronization for GitLab issues."""
+    if kind != ObjectKind.ISSUE:
+        logger.warning(f"Unexpected kind {kind} for on_resync_issues")
+        return []
+
+    if not hasattr(ocean, 'gitlab_handler'):
+        logger.error("GitLab handler not initialized. Please check on_start function.")
+        return []
+
+
+    fetch_method = lambda: ocean.gitlab_handler.fetch_resources(ObjectKind.ISSUE)
+    items = await fetch_resource(fetch_method)
+    return items
+
+
+@ocean.on_start()
+async def on_start() -> None:
+    """Initialize the GitLab handler."""
+    private_token = ocean.integration_config.get('token')
+    if not private_token:
+        logger.error("GitLab Token not provided in configuration")
+        return
+
+    try:
+        ocean.gitlab_handler = GitlabHandler(private_token)
+        logger.info("GitLab integration started and handler initialized")
+
+        await setup_webhooks()
+    except Exception as e:
+        logger.error(f"Failed to initialize GitLab handler: {str(e)}")
+
+async def setup_webhooks():
+    """Set up webhooks for the GitLab integration."""
+    events = ["push", "merge_requests", "issues"]
+
+
+    if not WEBHOOK_SECRET:
+        logger.error("Webhook secret not provided in configuration")
+        return
+
+
+    try:
+        await ocean.gitlab_handler.setup_webhooks(WEBHOOK_URL, WEBHOOK_SECRET, events)
+        logger.info("Webhooks set up successfully")
+    except Exception as e:
+        logger.error(f"Failed to set up webhooks: {str(e)}")
+
+
+@ocean.router.post("/webhook/gitlab")
+async def gitlab_webhook(request: Request):
+    """Handle incoming GitLab webhook events."""
+    if not WEBHOOK_SECRET:
```

Comment:
Do gitlab require webhook secrets? interrupting this because webhook secret is not set might not be the best